internal/ingress/controller/nginx.go

```diff
@@ -1084,37 +1074,65 @@ ratio = {{ .OtelSamplerRatio }}
 parent_based = {{ .OtelSamplerParentBased }}
 `
 
+func datadogOpentracingCfg(cfg ngx_config.Configuration) (string, error) {
+	jsn := map[string]interface{}{
+		"service": cfg.DatadogServiceName,
+		"agent_host": cfg.DatadogCollectorHost,
+		"agent_port": cfg.DatadogCollectorPort,
+		"environment": cfg.DatadogEnvironment,
+		"operation_name_override": cfg.DatadogOperationNameOverride,
+	}
+
+	// Omit "sample_rate" if the configuration's sample rate is unset (nil).
+	// Omitting "sample_rate" from the plugin JSON indicates to the tracer that
+	// it should use dynamic rates instead of a configured rate.
+	if cfg.DatadogSampleRate != nil {
+		jsn["sample_rate"] = *cfg.DatadogSampleRate
+	}
+
+	jsnBytes, err := json.Marshal(jsn)
```

Comment:
`b` or `buf` instead of `jsnBytes`. The name barely carries meaning because it's a forced placeholder with a 5 line lifecycle

Reply:
will do